### ISSUE TITLE
[Bug] Items in grouped CollectionView not positioned correctly on iOS

### ISSUE THREAD

**Issue by cristiproj** [xamarin/Xamarin.Forms#10436]
### Description
I am using grouped CollectionView with GridItemsLayout, with a span of 4.
The first time the page is loaded all the items are on the first column (screenshot 1). After trying to scroll, all the items are rearranged in the 4 column layout. But, the first element is incorrectly (screenshot 2).
The issue only happens on iOS. I tested in the simulator:
- iOS version: 12.2 everything is ok
- iOS version: 13.4 the broken behavior can be observed

### Steps to Reproduce

1. Create grouped CollectionView with griditemlayout
2. Bind to observablecollection / list
3. The behavior described can be observed

### Expected Behavior
The items are displayed correctly in the groupings according to the settings in the griditemlayout. With no awkward spaces.

### Actual Behavior
The items are not displayed correctly, not when the page is first loaded and not when scrolling 
### Basic Information

- Version with issue: 4.4.0.991265, 4.5.0.530 and 4.5.0.657
- Last known good version: - 
- IDE: Visual Studio Community 2019 for Mac, Version 8.5.4 (build 12)
- Platform Target Frameworks: 
  - iOS:  13.4
  - Android: 9.0
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Screenshots
Screenshot 1
![image](https://user-images.githubusercontent.com/63459725/80194760-89acc500-8623-11ea-866e-d41f42de79d4.png)
Screenshot 2
![image](https://user-images.githubusercontent.com/63459725/80194778-8fa2a600-8623-11ea-8a2a-ed333315d9c3.png)

### Reproduction Link

https://github.com/cristiproj/collectionviewissue

### Workaround

I have found found a possible workaround in using a listview with a datatemplate containting a flexlayout. Unfortunately, the items in the flexlayout are not arranged as nicely as in the collectionview. A small sample project can be found here: https://github.com/cristiproj/ListViewScheduleTest.

I think, another possible workaround would be by using a custom layout, but I did not look at it.

**Comment by jsuarezruiz** [comment]
@cristiproj Thanks for the reproduction sample.

**Comment by znelson32** [comment]
Same problem here, but my scenario is more simple. I have a grouped list of items with a GridItemsLayout. When I add Span="2" to get nice 2x2 boxes, the rendering is all screwed up. The items render over top of each other and jump around when scrolling.

**Comment by cristiproj** [comment]
I think the bug is also related to the iOS version. As I wrote in the issue, on iOS 12.2 the collectionview is rendered correctly, but not on 13.4.

**Comment by cristiproj** [comment]
A possible workaround is using a listview and flexlayout for the child items. Small sample project for testing: https://github.com/cristiproj/ListViewScheduleTest.

**Comment by znelson32** [comment]
Thank you, @cristiproj !

**Comment by kramer-e** [comment]
I have the same issue on the iPhone 8 emulator (iOS13.5) with a CollectionView that has a vertical GridItemsLayout with a span of 2. The items are stacked and glitched and move while scrolling. Issue #8738 from last year also describes this issue. Please give these grouping issues with the CollectionView some priority, I have opened issue #8383 in November last year and it is still not fixed.

**Comment by jeffdevalmeida** [comment]
I have the same problem. 
My code dont work in iPhone 11 and iPhone 12. Just in iPhone 8 and 8 Plus.

I tryied to use CollectionView then list my power bi reports. See this sample at iPhone 8 Plus (iOS 14.1):
![image](https://user-images.githubusercontent.com/54694777/97578199-3c911d80-19cf-11eb-9eff-91e6be7cb103.png)

But in iPhone 11 and 12, my list dont work. See this sample at iPhone 11 (iOS 14.1):
![image](https://user-images.githubusercontent.com/54694777/97578907-33548080-19d0-11eb-91b1-529d825c70ea.png)

After some analysis i found a possible solution for this case. I perceive that my tests is using the objects with same properties.
![image](https://user-images.githubusercontent.com/54694777/97579060-5e3ed480-19d0-11eb-935d-c10d38acf578.png)

I changed the properties... Now it's work!
New code with properties:
![image](https://user-images.githubusercontent.com/54694777/97579934-6d725200-19d1-11eb-8a64-8dbf15143ac6.png)

Working at iPhone 11 (iOS 14.1)
![image](https://user-images.githubusercontent.com/54694777/97580291-deb20500-19d1-11eb-91f6-1006124e85fa.png)

**Comment by demoh2019** [comment]
Update?

**Comment by demoh2019** [comment]
fixed with data loading (RemainingItemsThreshold)

**Comment by demoh2019** [comment]
@kramer-e 
could you find a solution?

**Comment by kramer-e** [comment]
No, but I've noticed that the issue doesn't appear with a span of 2 in the latest iOS + X.Forms version, maybe that was fixed in the new iOS version. The issue is still present with a span of 3 or more, along with other issues with grouping. I've used a workaround by creating a 2 dimensional array with a single column CollectionView, but that involves duplicate code and hurts performance.

Since this is an officially supported feature of the CollectionView, I thought this would be fixed in the X.Forms 5.0 release, but all the issues are still open unfortunately. I hope the Forms team can confirm that the grouping issues can be fixed in a service release this year.
